### PR TITLE
💡 update/correct method parameter documentation

### PR DESCRIPTION
**Pull request by vivekmore**
Update/Correct method parameter documentation

---

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/generator-base-private.js [hunk 1]
```diff
@@ -845,6 +847,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
    * Generate Entity Client Field Default Values
    *
    * @param {Array|Object} fields - array of fields
+   * @param clientFramework
```
Comment (vivekmore) [inline generators/generator-base-private.js:850]:
```suggestion
   * @param [clientFramework]
```

File: generators/generator-base-private.js [hunk 2]
```diff
@@ -908,6 +911,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
    * @param {Array|Object} fields - array of fields
    * @param {Array|Object} relationships - array of relationships
    * @param {string} dto - dto
+   * @param customDateType
```
Comment (vivekmore) [inline generators/generator-base-private.js:914]:
```suggestion
   * @param [customDateType]
```

File: generators/generator-base-blueprint.js [hunk 3]
```diff
@@ -483,7 +483,7 @@ module.exports = class JHipsterBaseBlueprintGenerator extends BaseGenerator {
    * Compose external blueprint module
    * @param {string} blueprint - name of the blueprint
    * @param {string} subGen - sub generator
-   * @param {any} options - options to pass to blueprint generator
+   * @param {any} extraOptions - options to pass to blueprint generator
```
Comment (vivekmore) [inline generators/generator-base-blueprint.js:486]:
```suggestion
   * @param {any} [extraOptions] - options to pass to blueprint generator
```

File: generators/generator-base-private.js [hunk 4]
```diff
@@ -1269,7 +1274,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
   /**
    * Returns the primary key value based on the primary key type, DB and default value
    *
-   * @param {string} primaryKeyType - the primary key type
+   * @param primaryKey
```
Comment (vivekmore) [inline generators/generator-base-private.js:1277]:
```suggestion
   * @param {string} primaryKey - the primary key type
```

File: generators/generator-base-private.js [hunk 5]
```diff
@@ -1559,7 +1564,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
    * Create a angular form path getter method of reference.
    *
    * @param {object} reference
-   * @param {string[]} valueDefinition
+   * @param prefix
```
Comment (vivekmore) [inline generators/generator-base-private.js:1567]:
```suggestion
   * @param {string[]} prefix
```

**Review by DanielFran — APPROVED** [review]
Trusting you in this

**Comment by vivekmore** [comment]
> Trusting you in this

👍🏼

**Comment by vivekmore** [comment]
👋🏼 @DanielFran - I'd like to suggest making `Rebase and Merge` a default PR merging strategy for following reasons:

1️⃣ - it makes the commit history linear
2️⃣ - linear commit history allows more readability/visualization of gradual discreet changes to a code base
3️⃣ - eliminates redundant merge commits that elongate the commit history
4️⃣ - allows for better traversing of the changes in the commit history when finding the source/origin of a bug
5️⃣ - allows better visualization to new contributors to understand what changes were applied in what order without having decipher changes in merge commits

🙏🏼 
Lastly - as a long time fan/supporter/contributor of JHipster, if this suggestion is acceptable, but requires additional efforts on the part of maintainers - I'm happy to take on that responsibility (provided I become the part of the team).
Once this approach yeilds an acceptable (if not better) commit history, we can think about [Rebase and merge via auto-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request)

Apologies for the unorthodox way to propose this change.

**Pull request merged**